src/main/kotlin/com/chattriggers/ctjs/minecraft/libs/renderer/Renderer.kt

```diff
@@ -225,91 +221,29 @@ object Renderer {
     }
 
     @JvmStatic
-    fun pushMatrix() = apply {
-        matrixStack.push()
+    @JvmOverloads
+    fun pushMatrix(stack: UMatrixStack = matrixStack) = apply {
+        matrixStackStack.addLast(stack)
+        matrixStack = stack
+        stack.push()
     }
 
     @JvmStatic
     fun popMatrix() = apply {
+        matrixStackStack.removeLast()
         matrixStack.pop()
     }
 
     @JvmStatic
-    fun translateToPlayer() = apply {
-        translate(-Client.camera.getX().toFloat(), -Client.camera.getY().toFloat(), -Client.camera.getZ().toFloat())
-    }
-
-    /**
-     * Begin drawing with the world renderer
-     *
-     * @param drawMode the GL draw mode
-     * @param vertexFormat The [VertexFormat] to use for drawing
-     * @return the Renderer to allow for method chaining
-     * @see com.chattriggers.ctjs.minecraft.libs.renderer.Shape.setDrawMode
-     */
-    @JvmStatic
-    @JvmOverloads
-    fun begin(
-        drawMode: DrawMode? = Renderer.drawMode,
-        vertexFormat: VertexFormat = VertexFormat.POSITION,
-    ) = apply {
-        pushMatrix()
-        enableBlend()
-        tryBlendFuncSeparate(770, 771, 1, 0)
-
-        worldRenderer.beginWithDefaultShader(drawMode?.toUC() ?: UGraphics.DrawMode.QUADS, vertexFormat.toUC())
-
-        firstVertex = true
-        began = true
-    }
-
-    /**
-     * Sets a new vertex in the Tessellator.
-     *
-     * @param x the x position
-     * @param y the y position
-     * @param z the z position
-     * @return the Tessellator to allow for method chaining
-     */
-    @JvmStatic
-    fun pos(x: Float, y: Float, z: Float) = apply {
```

Comment:
I think I'd like these functions to still exist for the 2d renderer and just delegate to the 3d renderer. They can be specialized if necessary (e.g. pos() should only take 2 params instead of 3).

Reply:
Should these all be copied over to Renderer? (pos, tex, color, normal, light, overlay).  My main concern is color, will we have to convert the existing color back to getColor?

Reply:
I think we should. The `getColor` change kinda sucks, but I think since `/ct migrate` exists and fixes that, it's not so bad